File: blog-dao/src/main/java/com/dao/UserMapper.java
package com.dao;

import com.entity.User;
import org.apache.ibatis.annotations.Param;

public interface UserMapper {
    int deleteByPrimaryKey(Integer uid);

    int insert(User record);

    int insertSelective(User record);

    User selectByPrimaryKey(Integer uid);

    int updateByPrimaryKeySelective(User record);

    int updateByPrimaryKey(User record);

    //按用户名查找
    User selectByUserName(String username);

    //对用户进行登陆验证
    User selectUser(@Param("username")String username, @Param("password") String password);

}

File: blog-dao/src/main/java/com/dao/UserMapper.kt
package com.dao

import com.entity.User
import org.apache.ibatis.annotations.Param

interface UserMapper {
    fun deleteByPrimaryKey(uid: Int?): Int

    fun insert(record: User): Int

    fun insertSelective(record: User): Int

    fun selectByPrimaryKey(uid: Int?): User

    fun updateByPrimaryKeySelective(record: User): Int

    fun updateByPrimaryKey(record: User): Int

    //按用户名查找
    fun selectByUserName(username: String): User

    //对用户进行登陆验证
    fun selectUser(@Param("username") username: String, @Param("password") password: String): User

}